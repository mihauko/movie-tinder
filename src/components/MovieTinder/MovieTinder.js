import React from 'react';

import MovieContext from './context';
import MovieHeader from './MovieHeader';
import MovieImg from './MovieImg';
import MovieDescription from './MovieDescription';
import MovieButtons from './MovieButtons';
import MovieSwipeWrapper from './MovieSwipeWrapper';


const MovieTinder = () => (
  <MovieContext.Consumer>
    {({ isLoadingData, filteredMovie, handleClick }) => {
      if (isLoadingData) return <div>Loading data..</div>;
      if (!filteredMovie) return <div>Sorry, there is not data.</div>;
      return (
        <MovieSwipeWrapper handleSwipe={handleClick}>
          <MovieHeader title={filteredMovie.title} rating={filteredMovie.rating} />
          <MovieImg title={filteredMovie.title} src={filteredMovie.imageURL} />
          <MovieDescription summary={filteredMovie.summary} />
          <MovieButtons handleClick={handleClick} />
        </MovieSwipeWrapper>
      );
    }}
  </MovieContext.Consumer>
);

export default MovieTinder;
