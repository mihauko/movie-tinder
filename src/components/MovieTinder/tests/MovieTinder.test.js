import React from 'react';
import { render } from '@testing-library/react';
import MovieContext from '../context';

import MovieTinder from '../MovieTinder';


describe('<MovieTinder>', () => {
  const mockedContext = {
    filteredMovie: {
      id: '1',
      // eslint-disable-next-line max-len
      imageURL: 'https://images-na.ssl-images-amazon.com/images/M/MV5BMTUzNTE2NTkzMV5BMl5BanBnXkFtZTgwMDAzOTUyMDI@._V1_SY1000_CR0,0,674,1000_AL_.jpg',
      title: 'Inferno',
      summary: 'Lorem ipsumâ€¦.',
      rating: 5.3,
    },
    handleClick: jest.fn(),
  };

  it('render correctly component', () => {
    const container = render(
      <MovieContext.Provider value={mockedContext}>
        <MovieTinder />
      </MovieContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
