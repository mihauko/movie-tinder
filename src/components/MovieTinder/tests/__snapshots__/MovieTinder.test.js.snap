// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieTinder> render correctly component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="sc-AxgMl cuLIma"
        id="swipeable"
      >
        <div
          class="sc-AxjAm AOfgU"
        >
          Inferno 5.3/10
        </div>
        <div
          class="sc-AxirZ dqVvws"
        >
          <img
            alt="Inferno"
            class="sc-AxiKw jipEWU"
            src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTUzNTE2NTkzMV5BMl5BanBnXkFtZTgwMDAzOTUyMDI@._V1_SY1000_CR0,0,674,1000_AL_.jpg"
            title="Inferno"
          />
        </div>
        <div
          class="sc-AxhCb iLrMEl"
        >
          Lorem ipsum….
        </div>
        <div
          aria-label="outlined secondary button group"
          class="MuiButtonGroup-root sc-AxhUy bKjbty MuiButtonGroup-fullWidth"
          role="group"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-containedPrimary MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Accept
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-containedSecondary MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Reject
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="sc-AxgMl cuLIma"
      id="swipeable"
    >
      <div
        class="sc-AxjAm AOfgU"
      >
        Inferno 5.3/10
      </div>
      <div
        class="sc-AxirZ dqVvws"
      >
        <img
          alt="Inferno"
          class="sc-AxiKw jipEWU"
          src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTUzNTE2NTkzMV5BMl5BanBnXkFtZTgwMDAzOTUyMDI@._V1_SY1000_CR0,0,674,1000_AL_.jpg"
          title="Inferno"
        />
      </div>
      <div
        class="sc-AxhCb iLrMEl"
      >
        Lorem ipsum….
      </div>
      <div
        aria-label="outlined secondary button group"
        class="MuiButtonGroup-root sc-AxhUy bKjbty MuiButtonGroup-fullWidth"
        role="group"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Accept
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-containedSecondary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Reject
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
